modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectDefinitionMVCActionCommand.java

```diff
@@ -66,37 +69,39 @@ protected void doProcessAction(
 
 		try {
 			_importObjectDefinition(actionRequest);
-
-			SessionMessages.add(
-				actionRequest, "importObjectDefinitionSuccessMessage");
-
-			hideDefaultSuccessMessage(actionRequest);
 		}
 		catch (Exception exception) {
 			if (_log.isDebugEnabled()) {
 				_log.debug(exception);
 			}
 
-			if (exception instanceof
-					ObjectDefinitionNameException.
-						MustBeginWithUpperCaseLetter ||
-				exception instanceof
-					ObjectDefinitionNameException.MustNotBeDuplicate ||
-				exception instanceof
-					ObjectDefinitionNameException.
-						MustOnlyContainLettersAndDigits) {
+			HttpServletResponse httpServletResponse =
+				_portal.getHttpServletResponse(actionResponse);
+
+			httpServletResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
 
-				SessionErrors.add(actionRequest, exception.getClass());
+			if (exception instanceof ObjectDefinitionNameException) {
+				Class<?> clazz = exception.getClass();
+
+				JSONPortletResponseUtil.writeJSON(
+					actionRequest, actionResponse,
+					JSONUtil.put(
+						"type",
+						"ObjectDefinitionNameException." +
+							clazz.getSimpleName()));
 			}
 			else {
-				SessionErrors.add(
-					actionRequest, "importObjectDefinitionErrorMessage");
+				JSONPortletResponseUtil.writeJSON(
+					actionRequest, actionResponse,
+					JSONUtil.put(
+						"title",
+						_language.get(
+							_portal.getHttpServletRequest(actionRequest),
+							"the-structure-was-not-successfully-imported")));
 			}
-
-			hideDefaultErrorMessage(actionRequest);
```

Comment:
Is it not necessary to hide default error message anymore?

Reply:
No because we don't use the `SessionErrors` anymore